winter/schema/query_parameters_inspector.py

```diff
@@ -1,54 +1,48 @@
-import inspect
+import typing
 from typing import List
 
-import docstring_parser
 from drf_yasg import openapi
 
 from .generation import get_argument_type_info
 from .method_arguments_inspector import MethodArgumentsInspector
-from .utils import update_doc_with_invalid_hype_hint
-from .. import type_utils
-from ..core import ComponentMethod
-from ..query_parameter import get_query_param_mapping
+from .type_inspection import TypeInfo
+from ..core import ComponentMethodArgument
+from ..query_parameter import QueryParameterAnnotation
 
+if typing.TYPE_CHECKING:  # pragma: no cover
+    from ..routing import Route
 
-class QueryParametersInspector(MethodArgumentsInspector):
 
-    def inspect_parameters(self, method: ComponentMethod) -> List[openapi.Parameter]:
-        docstring = docstring_parser.parse(method.func.__doc__)
-        params_docs = {param_doc.arg_name: param_doc for param_doc in docstring.params}
+class QueryParametersInspector(MethodArgumentsInspector):
+    _default_type_info = TypeInfo(openapi.TYPE_STRING)
 
+    def inspect_parameters(self, route: 'Route') -> List[openapi.Parameter]:
         parameters = []
 
-        for argument in method.arguments:
-            query_parameter_name = get_query_param_mapping(argument.method, argument.name)
-
-            if query_parameter_name is None:
-                continue
+        for argument, query_parameter_name in self._query_arguments(route):
 
-            param_doc = params_docs.get(query_parameter_name)
-            description = param_doc.description if param_doc else ''
-            default_params = {}
-            default = argument.parameter.default
-            if default is not inspect.Parameter.empty:
-                default_params['default'] = default
-            required = default is inspect.Parameter.empty and not type_utils.is_optional(argument.type_)
-            type_info = get_argument_type_info(argument)
+            default = argument.default if argument.has_default() else None
 
-            if type_info is None:
-                type_info_data = {'type': openapi.TYPE_STRING}
-                description = update_doc_with_invalid_hype_hint(description)
-            else:
-                type_info_data = type_info.as_dict()
+            type_info_data = get_argument_type_info(argument)
```

Comment:
Maybe rename `get_argument_type_info` to something `get_argument_info` to avoid messing up with TypeInfo. And assign the result to something similar (not type info, because it's not TypeInfo, more likely parameter_data)